core-bundle/src/Filesystem/VirtualFilesystem.php

```diff
@@ -269,10 +285,12 @@ private function doListContents(string $path, bool $deep, int $accessFlags): \Ge
         // Read from adapter, but enhance result with extra metadata on demand
         /** @var FilesystemItem $item */
         foreach ($this->mountManager->listContents($path, $deep) as $item) {
+            $path = $item->getPath();
+
             yield $item
-                ->withPath(Path::makeRelative($item->getPath(), $this->prefix))
+                ->withPath(Path::makeRelative($path, $this->prefix))
                 ->withExtraMetadata(
-                    fn () => $this->dbafsManager->getExtraMetadata($item->getPath())
+                    fn () => $this->dbafsManager->getExtraMetadata($path)
```

Comment:
In case you're wondering: These lines are pure refactoring (no change in what they do). This way `getPath()` is not executed twice and it's easier to understand which path we are meaning, namely the original one before adjusting it.